### DISCUSSION TITLE
Anybody Able To Get Gdiplus Working With V?

### DISCUSSION THREAD

**Discussion by Wajinn** [vlang/v#13400] [General]
Does anybody have any examples or experience with using Gdiplus with V?  
Know there are several substitutes, but would like to specifically see Gdiplus usage.

**Comment by spytheman** [comment]
Does it have a C API?

**Comment by spytheman** [comment]
Can you show a simple example of a C program using it?

**Comment by Wajinn** [comment]
Other languages have different libraries for using Gdiplus/GDI+, which is C++.  So there is really no comparison that I know of to show.  Interested in if possible or how to use V to access Gdiplus.  In addition, TCC doesn't include the GDI+ headers.  Even if you add them, all kinds of other errors comes along with them being C++.   Just trying to get GdiplusStartup to work is a problem.

It appears that Go dealt with this by using libgdiplus, which is a C based implementation of GDI+.  Was wondering if any V people had went down this road already and knew or have solved this.
```
struct C.GdiplusStartupInput 
{
	GdiplusVersion int
	DebugEventCallback int
	SuppressBackgroundThread bool
	SuppressExternalCodecs bool
}
fn C.GdiplusStartup...
```

**Comment by JalonSolov** [comment]
Not aware of anyone doing this with V, so far.  If libgdiplus works, you could interface to it with V, but you'll have to create the interface code yourself:  https://github.com/vlang/v/blob/master/doc/docs.md#calling-c-from-v

**Reply by Wajinn** [reply]
Yes, aware that might be the case.  Didn't want to re-invent the wheel, if somebody already had or were enjoying wheel barrels.

**Comment by JalonSolov** [comment]
https://github.com/vlang/awesome-v has the currently most complete list of modules people have done for V.  If it's not there, it likely hasn't been made public.  Sadly, some of what's there doesn't compile with the latest version of V... I think many people are waiting for v1.0 before they try to update.

**Reply by Wajinn** [reply]
Well, hopefully they will consider updating at version 0.3 (looks like this year), where the language and syntax will be relatively stable.

**Comment by withs** [comment]
Hey,

I never used it but microsoft provide a Flat API to use gdi+ in c:
[stackoverflow thread](https://stackoverflow.com/questions/5114591/how-to-use-gdi-in-c)
[msdn about flat api](https://docs.microsoft.com/en-gb/windows/win32/gdiplus/-gdiplus-graphics-flat?redirectedfrom=MSDN)
